src/modules/fixEmbedTwitterVideo/fixEmbedTwitterVideo.module.ts

```diff
@@ -0,0 +1,173 @@
+import { ActionRowBuilder, ButtonBuilder, ButtonStyle, MessageType } from 'discord.js';
+import { z } from 'zod';
+
+import { createModule } from '../../core/createModule';
+import { resolveCatch } from '../../helpers/resolveCatch.helper';
+
+const FXTwitterResponseSchema = z.object({
+  code: z.number(),
+  message: z.string(),
+  tweet: z.object({
+    media: z
+      .object({
+        videos: z
+          .array(
+            z
+              .object({
+                type: z.enum(['gif', 'video']),
+                url: z.string(),
+              })
+              .optional(),
+          )
+          .optional(),
+      })
+      .optional(),
+  }),
+});
+
+type URLMapping = {
+  pattern: RegExp;
+  replacement: string;
+};
+
+const modulePrefixButtonId = 'fixEmbedTwitterVideo-';
+const deleteBotAnswerPrefixButtonId = modulePrefixButtonId + 'deleteBotAnswer-';
+const deleteBotAnswerButtonId = deleteBotAnswerPrefixButtonId + '$authorIdMessage';
+const ignoreConfirmationButtonId = modulePrefixButtonId + 'ignoreBotButtons';
+
+const twitterUrlMappings: URLMapping[] = [
+  {
+    pattern: /https?:\/\/(mobile\.)?(twitter|x)\.com\/(\S+)\/status\/(\d+)/g,
+    replacement: 'https://fxtwitter.com/$3/status/$4',
+  },
+];
+
+const FXTwitterUrlMappings: URLMapping[] = [
+  {
+    pattern: /https?:\/\/fxtwitter\.com\/(\S+)\/status\/(\d+)/g,
+    replacement: 'https://api.fxtwitter.com/$1/status/$2',
+  },
+];
+
+const matchAndReplaceTweetLink = (message: string, urlMappings: URLMapping[]) => {
+  let tweetLink = '';
+
+  for (const urlMapping of urlMappings) {
+    const twitterLinks = message.match(urlMapping.pattern);
+
+    if (twitterLinks && twitterLinks.length > 0) {
+      tweetLink = twitterLinks[0].replace(urlMapping.pattern, urlMapping.replacement);
+
+      break;
+    }
+  }
+
+  return tweetLink;
+};
+
+const isTwitterVideo = async (tweetURL: string): Promise<boolean> => {
+  const twitterId = tweetURL.split('/').pop();
+  if (!twitterId) return false;
+
+  const apiFxTweetURL = matchAndReplaceTweetLink(tweetURL, FXTwitterUrlMappings);
+  const [tweetInfoError, tweetInfo] = await resolveCatch(fetch(apiFxTweetURL, { method: 'GET' }));
+
+  if (tweetInfoError) return false;
+
+  const tweetInfoJson = FXTwitterResponseSchema.safeParse(await tweetInfo.json());
```

Comment:
you should handle the possible error when reading the json

Reply:
the next line, i check if it's a success

Reply:
await .json can crash